pkg/installer/custominstallconfig_test.go

```diff
@@ -363,3 +374,54 @@ func verifyIgnitionFiles(t *testing.T, temp map[string]any, storageFiles []strin
 	}
 	assert.EqualValues(t, "ARO", config.Data["invoker"])
 }
+
+func verifyMasterPointerIgnition(t *testing.T, ignData []byte) {
+	ignContents := &igntypes.Config{}
+	err := json.Unmarshal(ignData, &ignContents)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	actualSource := *ignContents.Ignition.Config.Merge[0].Source
+	assert.EqualValues(t, expectedMasterIgnitionSource, actualSource, fmt.Sprintf("expected master pointer ignition to be %s but found %s", expectedMasterIgnitionSource, actualSource))
+}
+
+func verifyUpdateMCSCertKey(t *testing.T, bootstrap *bootstrap.Bootstrap) {
+	config := &igntypes.Config{}
+	config = bootstrap.Config
+
+	cert := &x509.Certificate{}
+	var data []byte
+
+	for i, fileData := range config.Storage.Files {
+		if fileData.Path == mcsCertFile {
+			contents := config.Storage.Files[i].Contents.Source
+			replaceable := "data:text/plain;charset=utf-8;base64,"
+			replaced := strings.Replace(*contents, replaceable, "", 1)
+
+			decodedCert, err := base64.StdEncoding.DecodeString(replaced)
```

Comment:
We could use `dataurl.DecodeString(contents)`, as we do elsewhere in the installer, to fully parse the URL and avoid the issue Aditya raised above.
Similarly `dataurl.EncodeBytes()` to create the data: URLs.